### ISSUE TITLE
Request: intelligent updates

### ISSUE THREAD

**Issue by sorenmaigaard** [Kemmey/Kemmey-TeslaWatch-Public#159]
Hi

Great app!

Suggestion: it is expensive battery wise to always wake the car to get a status. 
And you added some timing options for that. 

When it is most important to follow the updates is when the car is charging. That is also when it is free to wake the car and check. 

So perhaps an intelligent way to refresh status is to say:
- wake the car. Check status. 
- if the car is charging, keep checking the status every 5 minutes 
- when status becomes ‘not connected to charger anymore’ - check every hour until set time.
- at another set one, start checking again every hour
- when charging detected, go back to step 1

I think that would be a great way to handle that.

**Comment by Kemmey** [comment]
Already does that. 15 min intervals while charging. 5 min intervals while supercharging. 15 min intervals while driving. All others use configured interval options in settings.

**Comment by sorenmaigaard** [comment]
Awesome!

How does it know to switch between them?

Let’s say I have an interval of 1 hour set.
I now plug it in for charging. Would it take (up to) an hour for it to realize this and switch to the 15 min interval?
Or does it somehow find out (by polling without wake up perhaps?)

Best regards
Søren

On 4 Oct 2020, at 10.07, Kim Hansen <notifications@github.com> wrote:

﻿

Already does that. 15 min intervals while charging. 5 min intervals while supercharging. 15 min intervals while driving. All others use configured interval options in settings.

**Comment by Kemmey** [comment]
It changes as it see the states change. So exactly as in your example. Up to one hour to detect and then new schedule until charged and state changes again and new schedule is set. You can always just tap the complication and launch the app to refresh immediately. So if you just plugged in and want to keep track, just launch the app once to get the quicker interval. I find that to be the best compromise between up to date data and power usage on both watch and vehicle.

**Comment by sorenmaigaard** [comment]
Understood.

If I set the interval in settings to a low value, but don’t allow the app to wake up the car, would it detect that the car is charging/driving/supercharging quickly (because it is then awake) automatically?

Naturally I would then get very few updates when the car is parked and not charging, since the app would not wake up the car here and get a status.

Best
Søren

On 4 Oct 2020, at 10.46, Kim Hansen <notifications@github.com> wrote:

﻿

It changes as it see the states change. So exactly as in your example. Up to one hour to detect and then new schedule until charged and state changes again and new schedule is set. You can always just tap the complication and launch the app to refresh immediately. So if you just plugged in and want to keep track, just launch the app once to get the quicker interval. I find that to be the best compromise between up to date data and power usage on both watch and vehicle.

**Comment by Kemmey** [comment]
Yes, but be aware that watchOS will only allow ~50 background updates per 24 hours. So you would be hitting the background update budget limit quite quickly doing so.

**Comment by sorenmaigaard** [comment]
Hi

Interesting.
Is that per app or total background updates across apps.

Running every 15 min from 7-19 results in 48 updates. That could work 🙂

Best
Søren

On 4 Oct 2020, at 12.19, Kim Hansen <notifications@github.com> wrote:

﻿

Yes, but be aware that watchOS will only allow ~50 background updates per 24 hours. So you would be hitting the background update budget limit quite quickly doing so.

**Comment by Kemmey** [comment]
It is per app :-D so yeah, it would work for a normal workday. But you'd have to turn your watch off or manually adjust the schedule each morning and evening. 

I suggest just tapping the complication once, when you want to monitor "intensely". My experience is that I do so anyway...
